website/static/js/comment.js

```diff
@@ -132,29 +132,61 @@ BaseComment.prototype.setupToolTips = function(elm) {
     });
 };
 
+BaseComment.prototype.fetchUserData = function() {
+    var self = this;
+    var request = osfHelpers.ajaxJSON(
+        'GET',
+        osfHelpers.apiV2Url('users/me/', {}),
+        {'isCors': true});
+    request.done(function(response) {
+        self.author = {
+            'id': response.data.id,
+            'url': response.data.links.html,
+            'name': response.data.attributes.full_name,
+            'gravatarUrl': response.data.links.profile_image
+        };
+        return self.author;
+    });
+};
+
 BaseComment.prototype.fetch = function() {
     var self = this;
     var deferred = $.Deferred();
     if (self._loaded) {
         deferred.resolve(self.comments());
     }
-    $.getJSON(
-        nodeApiUrl + 'comments/',
-        {target: self.id()},
-        function(response) {
-            self.comments(
-                ko.utils.arrayMap(response.comments.reverse(), function(comment) {
-                    return new CommentModel(comment, self, self.$root);
-                })
-            );
-            self.unreadComments(response.nUnread);
-            deferred.resolve(self.comments());
-            self._loaded = true;
-        }
-    );
+    var url = osfHelpers.apiV2Url('nodes/' + window.contextVars.node.id + '/comments/', {query: 'embed=user'});
+    if (self.id() !== undefined) {
+        url = osfHelpers.apiV2Url('comments/' + self.id() + '/replies/', {query: 'embed=user'});
+    }
+    var request = osfHelpers.ajaxJSON(
+        'GET',
+        url,
+        {'isCors': true});
+    request.done(function(response) {
+        self.comments(
+            ko.utils.arrayMap(response.data, function(comment) {
+                return new CommentModel(comment, self, self.$root);
+            })
+        );
+        setUnreadCommentCount(self);
+        deferred.resolve(self.comments());
+        self._loaded = true;
+    });
     return deferred;
 };
 
+var setUnreadCommentCount = function(self) {
```

Comment:
You shouldn't need to pass in `self` to a function. Instead, make this a method.